.github/workflows/main.yml

```
@@ -0,0 +1,69 @@
+name: Build & Release
+
+on:
+  # Triggers the workflow on every pull request to master branch
+  pull_request:
+    branches:
+      - master
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: Android/MSTG-Android-Kotlin-App
+
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-${{ hashFiles('**/*.gradle*') }}-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/buildSrc/**/*.kt') }}
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+
+      - name: Build the app
+        run: ./gradlew assembleRelease
+        
+      - name: List Downloaded artifacts (sampling of download-artifact)
+        run: ls -l 
+        
+      - name: Rename build 
+        run: mv ./app/build/outputs/apk/release/app-release-unsigned.apk ./app/build/outputs/apk/release/MASTG-Android.apk
+
+      - name: Upload apk
+        uses: actions/upload-artifact@v2
+        with:
+          name: app-release
+          path: ./Android/MSTG-Android-Kotlin-App/app/build/outputs/apk/release/MASTG-Android.apk
+      
+  release:
+    runs-on: ubuntu-latest
+    needs: [build]
+    if: startsWith(github.ref, 'refs/tags/')
+    steps:
+      - uses: actions/download-artifact@v2
+      - name: List Downloaded artifacts (sampling of download-artifact)
+        run: ls -l 
+
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        with:
+          prerelease: false
+          draft: true
+          generate_release_notes: true
+          files: |
+            app-release/*.apk
```

would this work do so that we don't have a zip file as the artifact and have the apk instead?
```suggestion
            app-release/MASTG-Android.apk
```

---

Hey @cpholguera  I've added a step to sign the release. An APK is created on the release step (you might be looking at the GitHub Actions artifact which is always compressed as a zip file?). Have a look at an example release here: https://github.com/jsherm-fwdsec/MASTG-Hacking-Playground/releases/tag/0.9.0